### PR TITLE
Droid 1602 test merge main

### PR DESCRIPTION
**Pull request by uburoiubu**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: app/src/main/java/com/anytypeio/anytype/di/main/NetworkModeModule.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import javax.inject.Named
+import javax.inject.Singleton
+
+@Module
+object NetworkModeModule {
+
+    @JvmStatic
+    @Provides
+    @Singleton
+    @Named("network_mode")
```
Comment (uburoiubu) [inline app/src/main/java/com/anytypeio/anytype/di/main/NetworkModeModule.kt:19]: consider exporting to constant

File: core-models/src/main/java/com/anytypeio/anytype/core_models/Command.kt [hunk 2]
```diff
@@ -5,6 +5,21 @@ import com.anytypeio.anytype.core_models.primitives.TypeKey
...
+        val avatarPath: String?,
+        val icon: Int,
+        val networkMode: NetworkMode? = null,
+        val customConfigFilePath: String? = null
+    ) : Command()
+
+    data class AccountSelect(
+        val id: String,
+        val path: String,
+        val networkMode: NetworkMode? = null,
```
Comment (uburoiubu) [inline core-models/src/main/java/com/anytypeio/anytype/core_models/Command.kt:19]: `NetworkMode = NetworkMode.Default`

File: core-models/src/main/java/com/anytypeio/anytype/core_models/Command.kt [hunk 3]
```diff
@@ -5,6 +5,21 @@ import com.anytypeio.anytype.core_models.primitives.TypeKey
...
+        val customConfigFilePath: String? = null
```
Comment (uburoiubu) [inline core-models/src/main/java/com/anytypeio/anytype/core_models/Command.kt:20]: `networkConfigFilePath`

File: core-models/src/main/java/com/anytypeio/anytype/core_models/NetworkMode.kt [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+data class NetworkModeConfig(
+    val networkMode: NetworkMode = NetworkMode.DEFAULT,
+    val userFilePath: String? = null,
+    val storedFilePath: String? = null
+)
+
+object NetworkModeConstants {
+    const val NETWORK_MODE_PREF = "pref.network_mode"
```
Comment (uburoiubu) [inline core-models/src/main/java/com/anytypeio/anytype/core_models/NetworkMode.kt:19]: let's encapsulate these keys for shared preferences in `Persistence `module

File: domain/src/main/java/com/anytypeio/anytype/domain/auth/interactor/CreateAccount.kt [hunk 5]
```diff
@@ -12,25 +14,32 @@ import com.anytypeio.anytype.domain.platform.MetricsProvider
 open class CreateAccount(
     private val repository: AuthRepository,
     private val configStorage: ConfigStorage,
```
Comment (uburoiubu) [inline domain/src/main/java/com/anytypeio/anytype/domain/auth/interactor/CreateAccount.kt:16]: please add `// TODO rename config storage`

File: presentation/src/main/java/com/anytypeio/anytype/presentation/onboarding/signup/OnboardingVoidViewModel.kt [hunk 6]
```diff
@@ -80,16 +80,14 @@ class OnboardingVoidViewModel @Inject constructor(
 
     private fun proceedWithCreatingAccount() {
```
Comment (uburoiubu) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/onboarding/signup/OnboardingVoidViewModel.kt:81]: please add `Deprecated("To be deleted")` to this view model constructor

File: core-models/src/main/java/com/anytypeio/anytype/core_models/NetworkMode.kt [hunk 7]
```diff
@@ -0,0 +1,26 @@
+package com.anytypeio.anytype.core_models
+
+import com.anytypeio.anytype.core_models.NetworkModeConstants.NETWORK_MODE_CUSTOM
+import com.anytypeio.anytype.core_models.NetworkModeConstants.NETWORK_MODE_DEFAULT
+import com.anytypeio.anytype.core_models.NetworkModeConstants.NETWORK_MODE_LOCAL
+
+
+enum class NetworkMode(val value: String) {
```
[the remark this answers is not in the document]
Comment (konstantiniiv) [reply core-models/src/main/java/com/anytypeio/anytype/core_models/NetworkMode.kt:8]: deleted

File: core-models/src/main/java/com/anytypeio/anytype/core_models/NetworkMode.kt (hunk 4 above)
Comment (konstantiniiv) [reply core-models/src/main/java/com/anytypeio/anytype/core_models/NetworkMode.kt:19]: moved to NetworkModeProvider

File: presentation/src/main/java/com/anytypeio/anytype/presentation/onboarding/signup/OnboardingVoidViewModel.kt (hunk 6 above)
Comment (konstantiniiv) [reply presentation/src/main/java/com/anytypeio/anytype/presentation/onboarding/signup/OnboardingVoidViewModel.kt:81]: done

File: core-models/src/main/java/com/anytypeio/anytype/core_models/Command.kt (hunk 3 above)
Comment (konstantiniiv) [reply core-models/src/main/java/com/anytypeio/anytype/core_models/Command.kt:20]: done

File: app/src/main/java/com/anytypeio/anytype/di/main/NetworkModeModule.kt (hunk 1 above)
Comment (konstantiniiv) [reply app/src/main/java/com/anytypeio/anytype/di/main/NetworkModeModule.kt]: done

File: core-models/src/main/java/com/anytypeio/anytype/core_models/Command.kt (hunk 2 above)
Comment (konstantiniiv) [reply core-models/src/main/java/com/anytypeio/anytype/core_models/Command.kt]: done

**Review by konstantiniiv — APPROVED** [review]

**Pull request merged**